treenode.go

```diff
@@ -398,7 +405,7 @@ func (n *TreeNodeInstance) dispatchHandler(msgSlice []*ProtocolMsg) error {
 	return nil
 }
 
-func (n *TreeNodeInstance) reflectCreate(t reflect.Type, msg *ProtocolMsg) reflect.Value {
+func (n *TreeNodeInstance) reflectCreate(t reflect.Type, msg *ProtocolMsg) (reflect.Value, error) {
```

Comment:
i feel it's not 100% clean to do this check inside `reflectCreate` because the function name doesn't tell the programmer what it's doing accurate. maybe it's better to change the function name to reflect its now semantics e.g., `createValueAndVerify`, or move the verification logic into a new function.